eth-custodian/contracts/EthCustodian.sol

```diff
@@ -119,7 +122,7 @@ contract EthCustodian is ProofKeeper, AdminControlled {
             result.ethCustodian == address(this),
             'Can only withdraw coins that were expected for the current contract'
         );
-        payable(result.recipient).transfer(result.amount);
+        payable(result.recipient).call{value: result.amount}("");
```

Comment:
Catch the result and revert in case of failure, this way the proof is only consumed if the funds were sent successfully.

```suggestion
        (bool success, ) = payable(result.recipient).call{value: result.amount}("");
        require(success);
```

Reply:
+ to what @mfornet mentioned, should add `assert(success)` before the event.

Reply:
Fixed in a387de7